pandas/core/internals/array_manager.py

```diff
@@ -339,9 +341,24 @@ def where(self, other, cond, align: bool, errors: str, axis: int) -> ArrayManage
             axis=axis,
         )
 
-    # TODO what is this used for?
-    # def setitem(self, indexer, value) -> ArrayManager:
-    #     return self.apply_with_block("setitem", indexer=indexer, value=value)
+    def setitem(self, indexer, value, column_idx) -> ArrayManager:
```

Comment:
Block.setitem isn't always inplace, its a try-inplace-fallback-to-cast

Reply:
Yep, but not in ArrayManager (I know I need to open an issue about that to discuss this in general)